src/benchmarks/micro/libraries/System.Net.Sockets/Perf.Socket.cs

```diff
@@ -116,6 +117,43 @@ public async Task ReceiveAsyncThenSendAsync_SocketAsyncEventArgs()
             }
         }
 
+        [Benchmark]
+        public async Task ReceiveFromAsyncThenSendToAsync_Task()
+        {
+            ReadOnlyMemory<byte> clientBuffer = new byte[1];
+            Memory<byte> serverBuffer = new byte[1];
+
+            EndPoint ep = new IPEndPoint(IPAddress.None, 0);
+
+            for (int i = 0; i < InnerIterationCount; i++)
+            {
+                ValueTask<SocketReceiveFromResult> r = _udpServer.ReceiveFromAsync(serverBuffer, SocketFlags.None, ep);
+                await _udpClient1.SendToAsync(clientBuffer, SocketFlags.None, _udpServer.LocalEndPoint);
+                await r;
+                r = _udpServer.ReceiveFromAsync(serverBuffer, SocketFlags.None, ep);
+                await _udpClient2.SendToAsync(clientBuffer, SocketFlags.None, _udpServer.LocalEndPoint);
+                await r;
```

Comment:
Do we need to be concerned about packet loss causing the test to hang?

Reply:
I would hope not but time will tell. With the data segment small (and I purposely let it be 1) the seer should always fit into socket buffer. And since there is read before next write I see no possibility for buffer contention. Since there is no real network involved I see no reason for packet loss.

I can set cancellation token (or Wait) just in case and set timeout in the sync version if you feel it is worth it. 
(We often do that in functional tests)